src/tools/cephfs/cephfs-shell

```diff
@@ -1124,9 +1125,11 @@ sub-directories, files')
                 for i in reversed(sorted(set(dirwalk(dir_)))):
                     i = os.path.normpath(i)
                     try:
-                        xattr = cephfs.getxattr(to_bytes(i), 'ceph.dir.rbytes')
-                        self.poutput('{:10s} {}'.format(
-                            humansize(int(xattr.decode('utf-8'))), '.' + i))
+                        if stat.S_ISDIR(cephfs.stat(i_path):
+                            xattr = cephfs.getxattr(to_bytes(i),
+                                                    'ceph.dir.rbytes')
+                            self.poutput('{:10s} {}'.format(humansize(int(
+                                xattr.decode('utf-8'))), '.' + i))
```

Comment:
No, the right thing to do here would be to just statx the file and print its disk usage (st.stx_blocks * st.stx_blksize).

Reply:
I thought I had commented this but evidently I did forget.

I am using `st.st_size` instead of `st.stx_blocks * st.stx_blksize`. `st.stx_blocks * st.stx_blksize` led to values that were much higher than GNU du reported.

I've also noticed that GNU du ignores non-directory files unless they are passed explicitly. Do we want to mimic this?

```
$ du
28	./a
28	./c
28	./d
28	./b
144	.
$ ls
a  b  c  d  hlink1  slink1  slink2  SubmittingPatches.rst
$ du slink1 slink2 hlink1  SubmittingPatches.rst 
0	slink1
0	slink2
24	hlink1
24	SubmittingPatches.rst

```

Reply:
> I am using st.st_size instead of st.stx_blocks * st.stx_blksize. st.stx_blocks * st.stx_blksize led to values that were much higher than GNU du reported.

For regular files, st.st_size is correct. The difficulty with st.stx_blksize is that it's hardcoded to 4MB IIRC and that isn't actually the minimum blksize. i.e. the file (in CephFS) doesn't use blocks of disk in increments of 4MB. So for symlink,s you could just print the size of the link target string. For directories, maybe just the number of dentries x 1024 (as an approximation).

As far as the behavior of `du` with only printing directories, that is an okay default. Differences in behavior should be minimized.